### DISCUSSION TITLE
Remove selected items from Select in Repeater

### DISCUSSION THREAD

**Discussion by ALSHIHAB** [filamentphp/filament#8561] [Help] [answered]
I am creating a product and adding 5 different sections  from dropdown list in Repeater. Once I select the section from the list I need it to be hidden when create new section.

**Comment by leandrocfe** [comment]
Can you explain more about what you want to achieve? What code did you try?

**Reply by ALSHIHAB** [reply]
Ok. I have header section, description section, details dection, and so on. In repeater I have section dropdown and section content. When I select header section from drop down and fill content. I click add new section. I want in this new section to show me drop down of all sections except the selected header section. TO PREVENT SECTION DUPLICATION

**Reply by leandrocfe** [reply, marked as answer]
There is a [feature](https://github.com/orgs/filamentphp/projects/2/views/14?filterQuery=-status%3ADone+label%3Aenhancement+distinct&pane=issue&itemId=29696638) opened

A workaround

```php
Repeater::make('repeater')
    ->schema([
        Select::make('select')
            ->options([
                1 => 'Option 1',
                2 => 'Option 2',
                3 => 'Option 3',
            ])
            ->rules([
                function ($component) {
                    return function (string $attribute, $value, Closure $fail) use ($component) {

                        $items = $component->getContainer()->getParentComponent()->getState();
                        $selected = array_column($items, $component->getName());

                        if (count(array_unique($selected)) < count($selected)) {
                            $fail('You can only select one option.');
                        }
                    };
                },
            ])
    ]),
```

**Reply by ALSHIHAB** [reply]
Perfect. Thank you very much

**Reply by lloricode** [reply]
Thanks to this, now we can use official `distinct()` by filament https://github.com/filamentphp/filament/pull/8635
